From 7b08076384539e0aa42e174f0c934a1c3bfaeeaf Mon Sep 17 00:00:00 2001
From: Yu Chien Peter Lin <peterlin@andestech.com>
Date: Thu, 14 Mar 2024 10:50:10 +0800
Subject: [PATCH 13/17] riscv: plat-virt: Increase the TDDRAM_SIZE and adjust
 it to fit within the NAPOT region

OP-TEE OS core & TA  : 0xF100_0000 - 0xF1FF_FFFF (16MiB)
Non-secure static SHM: 0xF200_0000 - 0xF21F_FFFF ( 2MiB)

The associated device-tree node:

  opensbi-domains {
  	compatible = "opensbi,domain,config";

  	tmem: tmem {
  		compatible = "opensbi,domain,memregion";
  		base = <0x0 0xF1000000>;
  		order = <24>;
  	};
  ...
  };
---
 core/arch/riscv/plat-virt/conf.mk | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/arch/riscv/plat-virt/conf.mk b/core/arch/riscv/plat-virt/conf.mk
index 16579b22d..381d5f627 100644
--- a/core/arch/riscv/plat-virt/conf.mk
+++ b/core/arch/riscv/plat-virt/conf.mk
@@ -42,9 +42,9 @@ CFG_RISCV_WITH_M_MODE_SM ?= y
 supported-ta-targets = ta_rv64
 
 # Memory layout flags
-CFG_TDDRAM_START ?= 0x8e000000
-CFG_TDDRAM_SIZE  ?= 0x00f00000
+CFG_TDDRAM_START ?= 0xF1000000
+CFG_TDDRAM_SIZE  ?= 0x01000000
 $(call force,CFG_CORE_RESERVED_SHM,y)
-CFG_SHMEM_START  ?= 0x88f00000
+CFG_SHMEM_START  ?= 0xF2000000
 CFG_SHMEM_SIZE   ?= 0x00200000
 CFG_TEE_RAM_VA_SIZE ?= 0x00200000
-- 
2.34.1

